components/Code/index.jsx

```diff
@@ -38,16 +40,38 @@ CopyCode.propTypes = {
 };
 
 function Code(props) {
-  const { children, className, copyButtons, lang, meta, theme } = props;
+  const { children, className, copyButtons, lang, meta } = props;
 
   const langClass = className.search(/lang(?:uage)?-\w+/) >= 0 ? className.match(/\s?lang(?:uage)?-(\w+)/)[1] : '';
   const language = canonicalLanguage(lang) || langClass;
 
   const codeRef = React.createRef();
+
+  const useMediaQuery = query => {
+    const [matches, setMatches] = useState(false);
+
+    useEffect(() => {
+      const matchQueryList = window.matchMedia(query);
+      if (matchQueryList.matches !== matches) setMatches(matchQueryList.matches);
+      const handleChange = e => setMatches(e.matches);
+      matchQueryList.addEventListener('change', handleChange);
+      return () => matchQueryList.removeEventListener('change', handleChange);
+    }, [matches, query]);
+
+    return matches;
+  };
+
+  const colorMode = document.querySelector('[data-color-mode]').getAttribute('data-color-mode');
+  const initialTheme = localStorage.getItem('color-scheme');
+  const userColorScheme = useMediaQuery('(prefers-color-scheme: dark)') ? 'dark' : 'light';
```

Comment:
This seems like a lot of logic to have in this component for retrieval the color scheme. Is there maybe a context to pull from? How does it get set in the main app?

Reply:
It's setup as a hook here: https://github.com/readmeio/readme/blob/next/packages/react/src/core/hooks/useMediaQuery.js

And used like so:
https://github.com/readmeio/readme/blob/cf0e4b9ac1fd6da9f93c941ed7b39f2ea5a8dc3e/packages/react/src/Hub/Header/components/TopNav/index.jsx#L69

Reply:
My inclination would be to setup a DarkModeContext or ColorSchemeContext in the main app. Then we'd need to setup a way to pass that into RDMD. Perhaps RDMD could setup up a stubbed version that's always set to light, unless it gets overridden?

Reply:
Yeah, that was a path I considered. We use RDMD in quite a few places and we'd need to implement the logic in those spots/remember to do so in the future—not that I **love** this take on it either.

Reply:
Update: Kelly and I synced and we moved it to its own hook file :)